packages/app/cypress/e2e/runner/support/spec-loader.ts

```diff
@@ -0,0 +1,78 @@
+export const shouldHaveTestResults = ({ passCount, failCount, pendingCount }) => {
+  passCount = passCount || '--'
+  failCount = failCount || '--'
+
+  cy.findByLabelText('Stats', { timeout: 10000 }).within(() => {
+    cy.get('.passed .num', { timeout: 10000 }).should('have.text', `${passCount}`)
+    cy.get('.failed .num', { timeout: 10000 }).should('have.text', `${failCount}`)
+
+    if (pendingCount) {
+      cy.get('.pending .num', { timeout: 10000 }).should('have.text', `${pendingCount}`)
+    }
+  })
+}
+
+export type LoadSpecOptions = {
+  fileName: string
+  setup?: () => void
+  passCount?: number | string
+  failCount?: number | string
+  pendingCount?: number | string
+  hasPreferredIde?: boolean
+}
+
+export function loadSpec (options: LoadSpecOptions): void {
+  const {
+    fileName,
+    setup,
+    passCount = '--',
+    failCount = '--',
+    hasPreferredIde = false,
+    pendingCount,
+  } = options
+
+  cy.scaffoldProject('runner-e2e-specs')
+  cy.openProject('runner-e2e-specs')
+  cy.startAppServer()
+
+  cy.withCtx((ctx, options) => {
+    if (options.hasPreferredIde) {
+      // set preferred editor to bypass IDE selection dialog
+      ctx.coreData.localSettings.availableEditors = [
+        ...ctx.coreData.localSettings.availableEditors,
+        {
+          id: 'test-editor',
+          binary: '/usr/bin/test-editor',
+          name: 'Test editor',
+        },
+      ]
+
+      ctx.coreData.localSettings.preferences.preferredEditorBinary = 'test-editor'
```

Comment:
See [here for docs on how to update state](https://github.com/cypress-io/cypress/tree/10.0-release/packages/data-context#2-define-action-to-update-specs). I'm starting to understand the grand vision for the data-context, part of it is "a single place to update things" which is `this.ctx.update`. This will eventually let us have a log of all the actions and help with debugging, bug reports, etc.

Update `coreData` like this:

```ts
this.ctx.update(coreData => {
  coreData.localSettings.preferences.preferredEditorBinary = 'test-editor'
})
```


It doesn't really matter too much here, since it's just test code, but still good to start to enforce conventions.

Reply:
That works great too, thanks for the suggestion